src/core/thread/mle_router.cpp

```diff
@@ -1284,6 +1284,7 @@ otError MleRouter::HandleAdvertisement(const Message &         aMessage,
         if (processRouteTlv)
         {
             SuccessOrExit(error = ProcessRouteTlv(route));
+            aNeighbor = NULL; // aNeighbor is no longer valid after `ProcessRouteTlv`
```

Comment:
Turnout this is problematic, causing  `Cert_6_1_04_REEDAttachConnectivity.py` to fail

Reply:
Fixed: set `aNeighbor` to `NULL` only when it's in RouterTable.